composer/callbacks/checkpoint_saver.py

```diff
@@ -482,11 +589,19 @@ def _save_checkpoint(self, state: State, logger: Logger):
                         state.timestamp,
                     )
                     assert metadata_local_file_path is not None
-                    logger.upload_file(
-                        remote_file_name=metadata_remote_file_name,
-                        file_path=metadata_local_file_path,
-                        overwrite=self.overwrite,
-                    )
+                    if self.remote_uploader is not None:
+                        self.remote_uploader.upload_file_async(
+                            remote_file_name=metadata_remote_file_name,
+                            file_path=pathlib.Path(metadata_local_file_path),
+                            overwrite=self.overwrite,
+                        )
+                        local_remote_filenames.append(metadata_remote_file_name)
+                    else:
+                        logger.upload_file(
+                            remote_file_name=metadata_remote_file_name,
+                            file_path=metadata_local_file_path,
+                            overwrite=self.overwrite,
+                        )
```

Comment:
This seems very similar to below. Can you helper function it with remote_file_name and local_path?